### PR TITLE
[client] Return watcher errors from network monitor instead of reporting a network change

### PR DESCRIPTION
**Pull request by Optic00**
## Describe your changes

The network monitor's watcher goroutine (`checkChanges`) closed the shared `event` channel when the platform watcher returned an error. A closed channel made the `<-event` case in `Listen` fire in a busy loop until the debounce timer expired, so `Listen` returned `nil` — a watcher **failure** was reported as a **network change**, triggering an unnecessary client restart.

Changes:

- `checkChanges` no longer closes the `event` channel. It reports its error on a dedicated buffered `watchErrors` channel, guarded by `ctx.Done()` so the goroutine cannot block, and exits.
- `Listen` returns the watcher error wrapped (`watch network changes: %w`) instead of `nil`. On real cancellation it still returns `ctx.Err()`, preserving the existing clean-shutdown handling in the engine.
- If a watcher returns `context.Canceled` while the listener context is still active, `Listen` now returns the wrapped error instead of `nil`, so it cannot be mistaken for a detected network change.
- The default next-hop lookup is injectable, so the lifecycle tests no longer depend on host routes and run in restricted environments.
- New tests cover that the event channel is not closed, the error is delivered exactly once, `Listen` returns the wrapped watcher error, and the unexpected-cancellation edge case.

Verification: the new edge-case test fails against the previous implementation (`Listen() error = <nil>`) and passes with this change. The full package suite passes with `go test -race -count=1 ./client/internal/networkmonitor`, without requiring host routing access.

## Issue ticket number and link

N/A

## Stack

Standalone PR based on `main`.

### Checklist

- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [ ] Is a feature enhancement
- [ ] It is a refactor
- [x] Created tests that fail without the change (if possible)
- [x] This change does **not** modify the public API, gRPC protocols, functionality behavior, CLI / service flags, or introduce a new feature — **OR** I have discussed it with the NetBird team beforehand (link the issue / Slack thread in the description). See [CONTRIBUTING.md](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTING.md#discuss-changes-with-the-netbird-team-first).

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation

Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (internal bug fix in the client's network monitor; no user-facing behavior, API, or configuration changes)

### Docs PR URL (required if "docs added" is checked)

N/A

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved network monitoring error handling so watcher errors are reported reliably.
  * Preserved correct cancellation behavior when network watchers stop independently.
  * Prevented event channels from closing unexpectedly during monitoring.

* **Tests**
  * Added coverage for watcher errors, cancellation scenarios, and event-channel behavior.
  * Improved test synchronization for more reliable results.

### CODE REVIEW

File: client/internal/networkmonitor/monitor_test.go [hunk 1]
```diff
@@ -25,8 +41,12 @@ func (m *MocMultiEvent) checkChange(ctx context.Context, nexthopv4, nexthopv6 sy
...
 
 func TestNetworkMonitor_Close(t *testing.T) {
+	useTestNextHop(t)
+
+	watcherExited := make(chan struct{})
 	checkChangeFn = func(ctx context.Context, nexthopv4, nexthopv6 systemops.Nexthop) error {
 		<-ctx.Done()
+		close(watcherExited)
 		return ctx.Err()
 	}
```
[the remark this answers is not in the document]
Comment (Optic00) [reply client/internal/networkmonitor/monitor_test.go]: Fixed in 890af24a8. Every checkChangeFn override now goes through a helper that captures the previous value and restores it with t.Cleanup. A nested regression test covers restoration; go test -race ./client/internal/networkmonitor -count=1 passes.
